src/option-group.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const mixinSetters = require('./mixin-setters')
+const setValues = require('./set-values')
+
+class OptionGroup {
+
+  constructor (text) {
+    let values = {}
+    if (typeof text === 'object') {
+      values = text
+    } else if (typeof text === 'string') {
+      values.text = text
+    }
+
+    this._attachment = null
+    this.data = {}
+
+    setValues(this, values)
+    // Adds scope of `this` to each setter fn for chaining `.get()`
+    this.addSetterScopes()
+  }
+
+  attachment (attachment) {
+    this._attachment = attachment
+
+    return this
+  }
+
+  end () {
+    return this._attachment
+  }
+
+  json () {
+    return Object.assign({}, this.data)
+  }
+
+  toJSON () {
+    return this.json()
+  }
+}
+
+// props for Slack API - true gets a generic setter fn
+const PROPS = {
+  text: true,
+  option_groups: true,
+  option: function (text, value, description) {
+    if (value !== null && value !== undefined && typeof value === 'object') {
+      value = JSON.stringify(value)
+    }
+    let option = { text, value }
+    if (description !== undefined) option.description = description
+    if (!Array.isArray(this.data.option_groups)) {
+      this.data.option_groups = []
```

Comment:
this collection should probably be called `options` since it's the array of options, and that's the property name they need to be under for the Slack API